rag-sql-query-system/sql_router.py

```
@@ -0,0 +1,202 @@
+        if not tool_events:
+            return None
+
+        for tool_event in tool_events:
+            # append tool call chat messages to history
+            self.chat_history.append(tool_event.msg)
+
+        # # after all tool calls finish, pass input event back, restart agent loop
+        return InputEvent()
+
+async def main():
+    wf = RouterOutputAgentWorkflow(tools=[sql_tool, llama_cloud_tool], verbose=True, timeout=120)
```

Ruff (0.8.2)

Undefined name `sql_tool`

(F821)